azure/services/async/async.go

```diff
@@ -68,6 +68,34 @@ func processOngoingOperation(ctx context.Context, scope FutureScope, client Futu
 	return result, err
 }
 
+// getExistingResourceWithParameters is a helper that implements the logic for getting an a resource and constructing the parameters.
+func getExistingResourceWithParameters(ctx context.Context, scope FutureScope, client Creator, spec azure.ResourceSpecGetter, serviceName string) (interface{}, interface{}, error) {
```

Comment:
this is one of those situations where it might not make sense to have a helper function. This function is very specific to the CreateResource function above, takes all the same function parameters as input, and returns everything as output, which makes the code a bit awkward. It might make more sense if we were only returning the parameters as output, but here we also need the existing resource to use as result if parameters is nil. Consider moving the logic directly into `CreateResource`, or finding another way to breakout the code into smaller funcs if that makes the function too big / hard to read.